app/controllers/csvs_controller.rb

```diff
@@ -0,0 +1,107 @@
+class CsvsController < ApplicationController
+
+  rescue_from ActiveRecord::RecordNotFound do |exception|
+    respond_to do |format|
+      format.json { render json: {status: "not found"}.to_json, status: :not_found }
+      format.csv  { render body: "not found", status: :not_found }
+      format.html { render file: "#{Rails.public_path}/404.html", status: :not_found }
```

Comment:
what happens if you do `raise exception` for the html part ... does it render the normal 404 page ?